### DISCUSSION TITLE
How can CNN model take in images with dynamic sizes in ONNX format?

### DISCUSSION THREAD

**Discussion by AdvancedUno** [onnx/onnx#4733] [Q&A]
Hi! I am new to ONNX and I was trying to convert a Pytorch U-Net model into an Onnx format; however, I was having difficulty with putting in images with different sizes.
I know that when I create a model, I have to provide the expected input size for the model.
Is there any way for me to create a model that takes in images with random sizes?

**Comment by xadupre** [comment]
An ONNX model can leave the shape undefined (see https://onnx.ai/onnx/intro/python.html#a-simple-example-a-linear-regression). You should be able to use the following function to change the shapes on an existing model: https://onnx.ai/onnx/api/tools.html#update-inputs-outputs-dims.
